### PR TITLE
Py files and typechecking

### PR DESCRIPTION
**Pull request by knyazer**
Want to merge it @mikhail-vlasenko ?

### CODE REVIEW

**Comment by mikhail-vlasenko** [comment]
im not really done yet - some files don't have the annotations yet

**Comment by knyazer** [comment]
Remember we were talking about incorrect behavior of jaxtyping import hook? Can you please construct a minimal reproducible example, if it still does not work as expected?

**Comment by mikhail-vlasenko** [comment]
sure

**Comment by mikhail-vlasenko** [comment]
i suppose the _minimal_ example will then have new files and classes and so on. do you want it committed here?

**Comment by mikhail-vlasenko** [comment]
otherwise, the current version here is basically the minimal example. I also found that the wildcard import for `.wrappers` is not the issue

**Comment by knyazer** [comment]
Like, just write it as a code block, showing with comments the separation between files

File: ppo_brax_equinox/dataclasses.py [hunk 1]
```diff
@@ -0,0 +1,139 @@
...
+    # todo: fix with the new version of jaxtyping
+    steps_done: Union[int, Int32[Array, ""]] = eqx.field(default=0, converter=jnp.asarray)
+
+
+class TrainingState(eqx.Module):
+    optimizer: Optimizer
+    # todo: i get beartype.roar.BeartypeCallHintForwardRefException:
+    #  Forward reference "ppo_brax_equinox.dataclasses.ActionTanhConstraintWrapper" unimportable.
+    #  if i specify the class
+    agent: Any  # "BaseWrapper", or specifically "ActionTanhConstraintWrapper"
```
Comment (mikhail-vlasenko) [inline ppo_brax_equinox/dataclasses.py:138]: @knyazer maybe take a look at that

**Comment by mikhail-vlasenko** [comment]
@knyazer the minimal example:

```
------------ structure ------------
parent_dir/
├── test.py  (file that is ran)
└── ppo_brax_equinox/
    └── minimal_hook_example/
        ├── __init__.py
        ├── file1.py
        └── file2.py

------------ init ------------
from jaxtyping import install_import_hook


# decorate @jaxtyped and @beartype.beartype
with install_import_hook("ppo_brax_equinox.minimal_hook_example", "beartype.beartype"):
    from .file2 import bar

------------ file1 ------------
import jax.numpy as jnp
from jaxtyping import Array, Float


def foo() -> Float[Array, "2"]:  # wrong return shape
    return jnp.zeros(1)

------------ file2 ------------
import jax.numpy as jnp
from jaxtyping import Array, Float

from .file1 import foo


def bar() -> Float[Array, "1"]:
    return foo() + jnp.ones(1)

------------ test ------------
from ppo_brax_equinox.minimal_hook_example.file2 import bar

print(bar())  # raises
# beartype.roar.BeartypeCallHintReturnViolation: Function ppo_brax_equinox.minimal_hook_example.file1.foo() 
# return "Array([0.], dtype=float32)" violates type hint <class 'jaxtyping.Float[Array, '2']'>, 
# as <protocol "jaxlib.xla_extension.ArrayImpl"> "Array([0.], dtype=float32)" 
# not instance of <protocol "jaxtyping.Float[Array, '2']">.

```
the typechecking occurs for function `foo` even though it is not directly imported in the hook

**Pull request merged**